budgetportal/tests/test_infrastructure_projects.py

```diff
@@ -0,0 +1,100 @@
+from django.test import TestCase
+
+from budgetportal.models import InfrastructureProject
+
+
+class ProjectedExpenditureTestCase(TestCase):
+    """ Unit tests for get_projected_expenditure function """
+
+    def setUp(self):
+        self.fake_records = [
+            {'Financial Year': 2016, 'Amount': 100},
+            {'Financial Year': 2017, 'Amount': 100},
+            {'Financial Year': 2018, 'Amount': 100},
+            {'Financial Year': 2019, 'Amount': 100},
+            {'Financial Year': 2020, 'Amount': 100},
+            {'Financial Year': 2021, 'Amount': 100}
+        ]
+        self.project = InfrastructureProject()
+
+    def test_success(self):
+        self.project.records = self.fake_records
```

Comment:
this is a bit dodgy - is _setting_ `.records` intended to be a public interface? I reckon you might actually never want to set `InfraProject.records`, or at least mark it private as `_records`

you don't want to modify the class directly - you should instantiate it with the test data. it's not so much the test that's not right, as the fact that all the instantiation doesn't happen in the constructor.

An alternative is to extend the class with a class with the test data hardcoded and just used by `get_projected_expenditure` but that's very white-box.